In [1]:
# importing libraries
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

# reading in beer_reviews_data and storing it under variable df
df = pd.read_csv('data/beer_reviews_data.csv')

# displaying a random sample of the dataset utilizing the .sample() function
df.sample(n = 10, random_state = 42)

beer_ABV  beer_beerId  beer_brewerId  \
116087       5.8          234             76   
218685       8.7         1446            158   
63854       11.0        74491          16866   
234370       5.0        43341           1628   
501494       9.0        40187            132   
246729       9.0         2512            215   
489422       5.3          353            132   
243405      11.0         7975             48   
4124         8.4        52211          14879   
479954       5.9        17300           9529   

                                beer_name                beer_style  \
116087  Hacker-Pschorr Oktoberfest-M�rzen      M�rzen / Oktoberfest   
218685                    Hibernation Ale                   Old Ale   
63854                     4 Calling Birds   Belgian Strong Dark Ale   
234370                Southampton Altbier                   Altbier   
501494                    XS Imperial Red  American Amber / Red Ale   
246729       Chimay Grande R�serve (Blue)   Belgian Strong Dark Ale   
489422                       Mocha Porter           American Porter   
243405          Bi�re Du Boucanier Golden   Belgian Strong Pale Ale   
4124     Frog's Hollow Double Pumpkin Ale               Pumpkin Ale   
479954                           Pale Ale   American Pale Ale (APA)   

        review_appearance  review_palette  review_overall  review_taste  \
116087                4.0             3.5             4.0           4.5   
218685                4.5             4.5             4.0           3.5   
63854                 4.0             3.0             3.0           3.5   
234370                2.5             4.0             4.0           3.5   
501494                4.0             4.0             3.0           3.0   
246729                3.5             5.0             4.5           4.0   
489422                4.5             4.0             4.0           4.0   
243405                3.5             4.5             3.0           3.0   
4124                  4.0             3.5             3.0           3.5   
479954                4.5             4.0             3.5           4.0   

       review_profileName  review_aroma  \
116087        divineaudio           4.5   
218685          Cylinsier           3.0   
63854       womencantsail           3.0   
234370             DerekP           3.5   
501494         Gueuzedude           3.5   
246729             mentor           4.0   
489422           Stubbie1           4.0   
243405       Infamous7100           3.5   
4124          Huhzubendah           3.5   
479954         belgaridub           4.5   

                                              review_text  review_time  
116087  pint glass. a - pours a clear amber liquid wit...   1317698982  
218685  From a 12 oz. Pours a maple syrup brown that g...   1262818901  
63854   On-tap at the Bruery Almost black with a dense...   1321823635  
234370  According to the label, a "Dusseldorf-Style Br...   1280711924  
501494  Sampled on Tap at 1702, Tucson AZ; August 2008...   1219009004  
246729  Three finger dead tan head over a cloudy dark ...   1195591313  
489422  Appearance: Poured black, finished with a nice...   1041482821  
243405  Consumed this from a snifter, as you should wi...   1239405638  
4124    Bottle shared by Chaney. Thanks Brandon! A: Th...   1286322675  
479954  This came to the table a dark red-brown color....   1147897647

In [2]:
# understanding the data landscape of the dataset (dtype, non-null values and memory usage)
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 528870 entries, 0 to 528869
Data columns (total 13 columns):
 #   Column              Non-Null Count   Dtype  
---  ------              --------------   -----  
 0   beer_ABV            508590 non-null  float64
 1   beer_beerId         528870 non-null  int64  
 2   beer_brewerId       528870 non-null  int64  
 3   beer_name           528870 non-null  object 
 4   beer_style          528870 non-null  object 
 5   review_appearance   528870 non-null  float64
 6   review_palette      528870 non-null  float64
 7   review_overall      528870 non-null  float64
 8   review_taste        528870 non-null  float64
 9   review_profileName  528755 non-null  object 
 10  review_aroma        528870 non-null  float64
 11  review_text         528751 non-null  object 
 12  review_time         528870 non-null  int64  
dtypes: float64(6), int64(3), object(4)
memory usage: 52.5+ MB


In [3]:
# looking at descriptive statistics for all numericl columns
df.describe().transpose()

count          mean           std           min  \
beer_ABV           508590.0  7.017442e+00  2.204460e+00  1.000000e-02   
beer_beerId        528870.0  2.209847e+04  2.215828e+04  3.000000e+00   
beer_brewerId      528870.0  2.598423e+03  5.281805e+03  1.000000e+00   
review_appearance  528870.0  3.864522e+00  6.040104e-01  0.000000e+00   
review_palette     528870.0  3.758926e+00  6.853349e-01  1.000000e+00   
review_overall     528870.0  3.833197e+00  7.099618e-01  0.000000e+00   
review_taste       528870.0  3.765993e+00  6.690176e-01  1.000000e+00   
review_aroma       528870.0  3.817350e+00  7.189030e-01  1.000000e+00   
review_time        528870.0  1.224885e+09  7.605600e+07  8.843904e+08   

                            25%           50%           75%           max  
beer_ABV           5.300000e+00  6.500000e+00  8.500000e+00  5.770000e+01  
beer_beerId        1.745000e+03  1.436800e+04  4.052800e+04  7.731000e+04  
beer_brewerId      1.320000e+02  3.940000e+02  1.475000e+03  2.798000e+04  
review_appearance  3.500000e+00  4.000000e+00  4.000000e+00  5.000000e+00  
review_palette     3.500000e+00  4.000000e+00  4.000000e+00  5.000000e+00  
review_overall     3.500000e+00  4.000000e+00  4.500000e+00  5.000000e+00  
review_taste       3.500000e+00  4.000000e+00  4.000000e+00  5.000000e+00  
review_aroma       3.500000e+00  4.000000e+00  4.500000e+00  5.000000e+00  
review_time        1.174613e+09  1.240366e+09  1.288560e+09  1.326277e+09

In [4]:
# lastly understanding the dimensionality of the dataset
df.shape

(528870, 13)

### Dimensionality Results

+ This dataset contains 528,870 beer reviews
+ it has 13 columns 

### Another question

+ How many unique users make up the 528,870 reviews?

### Question 1: Which breweries produce the strongest beer?

#### Approach

+ In order to measure the strongest beers, we need to calculate the average ABV (Alcohol by Volume)
+ Therefore, group by the brewer ID, calculate the mean beer ABV and sort it in descending order
+ Using the .rank() function, then we can filter out for the topm 3 strongest Breweries

In [13]:
# grouping the dataset by brewer_abv and calculating the average alcohol by volume
df_abv = df.groupby('beer_brewerId')['beer_ABV'].mean().reset_index().sort_values(by = 'beer_ABV', ascending = False)

# rounding the average to the second decimal point and filling any null values with 0 for cleaner data
df_abv['beer_ABV'] = round(df_abv['beer_ABV'], 2).fillna(0)

# renaming the column for better readability
df_abv = df_abv.rename({"beer_ABV" : "avg_ABV"}, axis = 1)

# using the rank function to determine the top 3 breweries with the strongest beer
df_abv['rank'] = df_abv['avg_ABV'].rank(method = 'dense', ascending = False)

# displaying the dataset and filtering out for the top 3
df_abv[df_abv['rank'] <= 3]

beer_brewerId  avg_ABV  rank
784            6513    19.23   1.0
175             736    13.75   2.0
1644          24215    12.47   3.0